src/Stack/Setup.hs

```diff
@@ -1405,9 +1405,10 @@ installGHCPosix downloadInfo _ archiveFile archiveType tempDir destDir = do
 
     dir <- expectSingleUnpackedDir archiveFile tempDir
 
+    gccEnvMaybe <- fmap (\gcc -> Map.fromList [("CC", T.pack (toFilePath gcc))]) <$> (view $ configL.to configOverrideGccPath)
```

Comment:
That's... intense

Reply:
`configOverrideGccPath` is `Maybe (Path Abs File)`, `view` runs in `MonadReader`, so in `RIO`, hence we need two `fmap` to get to the fileapth. Once we have a filepath, we constract a `CC=gcc-filepath` env var from it

Reply:
I know, it's just a complexity spike wrt. surrounding code.

I'd take `views` instead of the outer `fmap`, so the code would zigzag less:

```haskell
gccEnv <- views (configL.to configOverrideGccPath) $
  maybe mempty $ \gcc ->
    Map.singleton "CC" (T.pack (toFilePath gcc))
```